find-address/src/main/java/com/esri/arcgisruntime/sample/findaddress/MainActivity.java

```diff
@@ -0,0 +1,309 @@
+        }
+      }
+    });
+  }
+
+  /**
+   * Geocode an address passed in by the user.
+   *
+   * @param address read in from searchViews
+   */
+  private void geoCodeTypedAddress(final String address) {
+    // check that address isn't null
+    if (address != null) {
+
+      // Execute async task to find the address
+      mLocatorTask.addDoneLoadingListener(new Runnable() {
+        @Override
+        public void run() {
+          if (mLocatorTask.getLoadStatus() == LoadStatus.LOADED) {
+            // Call geocodeAsync passing in an address
+            final ListenableFuture<List<GeocodeResult>> geocodeResultListenableFuture = mLocatorTask
+                .geocodeAsync(address, mAddressGeocodeParameters);
+            geocodeResultListenableFuture.addDoneListener(new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  // Get the results of the async operation
+                  List<GeocodeResult> geocodeResults = geocodeResultListenableFuture.get();
+                  if (geocodeResults.size() > 0) {
+                    displaySearchResult(geocodeResults.get(0));
+                  } else {
+                    Toast.makeText(getApplicationContext(), getString(R.string.location_not_found) + address,
+                        Toast.LENGTH_LONG).show();
+                  }
+                } catch (InterruptedException | ExecutionException e) {
+                  Log.e(TAG, "Geocode error: " + e.getMessage());
+                  Toast.makeText(getApplicationContext(), getString(R.string.geo_locate_error), Toast.LENGTH_LONG)
+                      .show();
+                }
+              }
+            });
+          } else {
+            Log.i(TAG, "Trying to reload locator task");
+            mLocatorTask.retryLoadAsync();
+          }
+        }
+      });
+      mLocatorTask.loadAsync();
+    }
+  }
+
+  /**
+   * Turns a GeocodeResult into Points and adds them to a GraphicOverlay which is then drawn on the map.
+   *
+   * @param geocodeResult a single geocode result
+   */
+  private void displaySearchResult(GeocodeResult geocodeResult) {
+    // dismiss any callout
+    if (mMapView.getCallout() != null && mMapView.getCallout().isShowing()) {
+      mMapView.getCallout().dismiss();
+    }
+    // clear map of existing graphics
+    mMapView.getGraphicsOverlays().clear();
+    mGraphicsOverlay.getGraphics().clear();
+    // create graphic object for resulting location
+    Point resultPoint = geocodeResult.getDisplayLocation();
+    Graphic resultLocGraphic = new Graphic(resultPoint, geocodeResult.getAttributes(), mPinSourceSymbol);
+    // add graphic to location layer
+    mGraphicsOverlay.getGraphics().add(resultLocGraphic);
+    // zoom map to result over 3 seconds
+    mMapView.setViewpointAsync(new Viewpoint(geocodeResult.getExtent()), 3);
+    // set the graphics overlay to the map
+    mMapView.getGraphicsOverlays().add(mGraphicsOverlay);
```

Comment:
We should add the `Callout` here when displaying a search result instead of forcing user to tap the marker to get the callout.

Reply:
broke showCallout() into showCallout() and identifyGraphic() to remove identification work from showCallout(). showCallout() now called when pin is drawn in displaySearchResult()